src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py

```diff
@@ -18,9 +19,9 @@ class ProjectBackfillSimilarIssuesEmbeddingsRecords(ProjectEndpoint):
     }
 
     def post(self, request: Request, project) -> Response:
-        if not features.has(
-            "projects:similarity-embeddings-backfill", project
-        ) or not is_active_superuser(request):
+        if not features.has("projects:similarity-embeddings-backfill", project) or (
+            not is_active_superuser(request) and not settings.SENTRY_SINGLE_ORGANIZATION
+        ):
```

Comment:
As I understand it, you're allowed to go through if any of these conditions are true. 
This would make it more obvious. 

```suggestion
        if not (
           features.has("projects:similarity-embeddings-backfill", project)
           or is_active_superuser(request)
           or settings.SENTRY_SINGLE_ORGANIZATION
        ):
```

Reply:
this is not quite right, but the conditional is confusing.

you need to have the flag, and either sentry_single_organization, or is_active_superuser. i can follow up with a rewrite of this conditional.

Reply:
let me make this less confusing

Reply:
kk, updated